file-share-service-desktop-client/Modules/Admin/JobViewModels/NewBatchJobViewModel.cs

```diff
@@ -192,13 +193,19 @@ protected internal override async Task OnExecuteCommand()
                 }
                 catch (Exception e)
                 {
-                    logger.LogError(e.Message);
-                    logger.LogInformation("File Share Service batch rollback started for batch ID:{BatchId}.", batchHandle.BatchId);                        
-                    CancellationTokenSource.Dispose();                        
-                    await fileShareClient.RollBackBatchAsync(batchHandle);
-                    logger.LogInformation("File Share Service batch rollback completed for batch ID:{BatchId}.", batchHandle.BatchId);
-                    ExecutionResult = e.ToString();
-                    throw;
+                    logger.LogError(e.ToString());
+
+                    if (batchHandle != null)
+                    {
+                        logger.LogInformation("File Share Service batch rollback started for batch ID:{BatchId}.", batchHandle?.BatchId);
+                        CancellationTokenSource.Dispose();
+                        await fileShareClient.RollBackBatchAsync(batchHandle);
+                        logger.LogInformation("File Share Service batch rollback completed for batch ID:{BatchId}.", batchHandle?.BatchId);
+                    }
+
+                    ExecutionResult = ((System.Net.Http.HttpRequestException)e).StatusCode == HttpStatusCode.BadRequest
```

Comment:
Please check the exception is actually an instance of `HttpRequestException` before doing this cast

Reply:
Removing this exception check as we are handling API exception at dot net client and from there will be returning DTO.
Thanks